diff --git a/starlette/_utils.py b/starlette/_utils.py
index b9b8639..fab4716 100644
--- a/starlette/_utils.py
+++ b/starlette/_utils.py
@@ -4,6 +4,7 @@ import functools
 import sys
 from collections.abc import AsyncGenerator, Awaitable, Callable, Generator
 from contextlib import AbstractAsyncContextManager, asynccontextmanager
+from inspect import iscoroutinefunction
 from typing import Any, Generic, Protocol, TypeVar, overload
 
 import anyio.abc
@@ -11,21 +12,10 @@ import anyio.abc
 from starlette.types import Scope
 
 if sys.version_info >= (3, 13):  # pragma: no cover
-    from inspect import iscoroutinefunction
     from typing import TypeIs
 else:  # pragma: no cover
-    from asyncio import iscoroutinefunction
-
     from typing_extensions import TypeIs
 
-if sys.version_info < (3, 11):  # pragma: no cover
-    try:
-        from exceptiongroup import BaseExceptionGroup
-    except ImportError:
-
-        class BaseExceptionGroup(BaseException):  # type: ignore[no-redef]
-            pass
-
 
 T = TypeVar("T")
 AwaitableCallable = Callable[..., Awaitable[T]]
