diff --git a/starlette/background.py b/starlette/background.py
index 55135ee..1ed7ddf 100644
--- a/starlette/background.py
+++ b/starlette/background.py
@@ -1,16 +1,15 @@
 from __future__ import annotations
 
-from collections.abc import Callable, Sequence
-from typing import Any, ParamSpec
-
 from starlette._utils import is_async_callable
 from starlette.concurrency import run_in_threadpool
 
-P = ParamSpec("P")
+if __debug__ and __import__("typing").TYPE_CHECKING:
+    from collections.abc import Callable, Sequence
+    from typing import Any
 
 
 class BackgroundTask:
-    def __init__(self, func: Callable[P, Any], *args: P.args, **kwargs: P.kwargs) -> None:
+    def __init__[**P](self, func: Callable[P, Any], *args: P.args, **kwargs: P.kwargs) -> None:
         self.func = func
         self.args = args
         self.kwargs = kwargs
@@ -27,7 +26,7 @@ class BackgroundTasks(BackgroundTask):
     def __init__(self, tasks: Sequence[BackgroundTask] | None = None):
         self.tasks = list(tasks) if tasks else []
 
-    def add_task(self, func: Callable[P, Any], *args: P.args, **kwargs: P.kwargs) -> None:
+    def add_task[**P](self, func: Callable[P, Any], *args: P.args, **kwargs: P.kwargs) -> None:
         task = BackgroundTask(func, *args, **kwargs)
         self.tasks.append(task)
 
